src/Microsoft.SqlTools.ServiceLayer/AutoParameterizaition/Exceptions/ParameterizationFormatException.cs

```diff
@@ -0,0 +1,41 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Microsoft.SqlTools.ServiceLayer.AutoParameterizaition.Helpers;
+using System;
+
+namespace Microsoft.SqlTools.ServiceLayer.AutoParameterizaition.Exceptions
+{
+    /// <summary>
+    /// ParameterizationFormatException is used to surface format exceptions encountered in the TSQL batch to perform auto-parameterization of literals
+    /// </summary>
+    class ParameterizationFormatException : FormatException
+    {
+        public readonly int LineNumber;
+        public readonly string LiteralValue;
+        public readonly string CodeSenseMessage;
+        public readonly string LogMessage;
+        public readonly string TelemetryMessage;
+        public readonly string SqlDatatype;
+        public readonly string CSharpDataType;
+
+        public ParameterizationFormatException(MessageHelper.MessageType type, string variableName, string sqlDataType, string cSharpDataType, string literalValue, int lineNumber)
+            : this(type, variableName, sqlDataType, cSharpDataType, literalValue, lineNumber, null)
+        {
+        }
+
+        public ParameterizationFormatException(MessageHelper.MessageType type, string variableName, string sqlDataType, string cSharpDataType, string literalValue, int lineNumber, Exception e)
+            : base(MessageHelper.GetLocalizedMessage(type, variableName, sqlDataType, literalValue), e)
+        {
+            LineNumber = lineNumber;
+            LiteralValue = literalValue;
+            SqlDatatype = sqlDataType;
+            CSharpDataType = cSharpDataType;
+            CodeSenseMessage = this.Message;
+            LogMessage = MessageHelper.GetLocaleInvariantMessage(type, variableName, sqlDataType) + ", Literal Value: " + literalValue + ", On line: " + lineNumber + ", CSharp DataType: " + cSharpDataType;
+            TelemetryMessage = MessageHelper.GetLocaleInvariantMessage(type, variableName, sqlDataType);
```

Comment:
Is the variable name the generated variable name? I guess that's ok to send - although is it actually going to be useful to have that? In general we shouldn't send anything that isn't going to provide value so make sure everything here is useful in some way

Reply:
The variable name is the user declared variable name. For example the following query...
```sql
DECLARE @Number int = 'ABCDEFG'

SELECT * FROM [dbo].[Table]
WHERE [N] = @Number
GO
```
...will display the following message. If there are multiple variable/paramers then the variable name will identify which one is causing the problem.
`Unable to Convert @Number to a Microsoft.Data.SqlClient.SqlParameter object. The specified literal cannot be converted to Int(System.Data.SqlDbType)., Literal Value: ABCDEFG, On line: 2, CSharp DataType: Int32`

Reply:
Ok. My default answer to anything user-generated is "no, we can't send that up as telemetry".

If having the name is important for the data you want to gather let's have a quick chat about what we can do.

You can still capture stuff like the types - just not anything user-generated. 

(not I'm not talking about the LogMessage here. That's fine as-is. I'm referring to the TelemetryMessage field)

Reply:
You know what? This `TelemetryMessage` is not even used anywhere now so the simplest thing to do would be to remove it alltogether.